File: app/models/qdm/risk_assessment.rb
module QDM
  # app/models/qdm/risk_assessment.rb
  class RiskAssessment < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :identifier
    field :basedOn
    field :parent
    field :status
    field :method
    field :code
    field :subject
    field :context
    field :occurrence
    field :condition
    field :performer
    field :reasonQDM::Code
    field :reasonReference
    field :basis
    field :prediction
    field :mitigation
    field :note
    field :qdmVersion, type: String, default: '4.0.0'
  end
end

File: app/assets/javascripts/RiskAssessment.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const RiskAssessmentSchema = DataElementSchema({
  identifier: Any,
  basedOn: Any,
  parent: Any,
  status: Any,
  method: Any,
  code: Any,
  subject: Any,
  context: Any,
  occurrence: Any,
  condition: Any,
  performer: Any,
  reasonCode: Any,
  reasonReference: Any,
  basis: Any,
  prediction: Any,
  mitigation: Any,
  note: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'RiskAssessment' },

});

module.exports.RiskAssessmentSchema = RiskAssessmentSchema;
module.exports.RiskAssessment = mongoose.model('RiskAssessment', RiskAssessmentSchema);
